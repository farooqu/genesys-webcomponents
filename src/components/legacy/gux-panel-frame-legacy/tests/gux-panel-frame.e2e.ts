import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-panel-frame', () => {
  const headerSlot = '<div slot="header"><h1>Header</h1></div>';
  const bodySlot = '<div slot="body"><p>Boby</p></div>';
  const footerSlot = '<div slot="footer"><footer>footer</footer></div>';
  const invalidSlot =
    '<div slot="invalid" id="invalid"><span>Invalid</span></div>';

  describe('#render', () => {
    [
      `<gux-panel-frame-legacy>${headerSlot}${bodySlot}${footerSlot}</gux-panel-frame-legacy>`,
      `<gux-panel-frame-legacy>${bodySlot}${footerSlot}</gux-panel-frame-legacy>`,
      `<gux-panel-frame-legacy>${headerSlot}${bodySlot}</gux-panel-frame-legacy>`,
      `<gux-panel-frame-legacy>${headerSlot}${footerSlot}</gux-panel-frame-legacy>`,
      `<gux-panel-frame-legacy>${headerSlot}</gux-panel-frame-legacy>`,
      `<gux-panel-frame-legacy>${headerSlot}${bodySlot}${invalidSlot}${footerSlot}</gux-panel-frame-legacy>`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-panel-frame-legacy');
        const elementShadowDom = await element.find(
          'pierce/.gux-panel-container'
        );

        expect(element.outerHTML).toMatchSnapshot();
        expect(elementShadowDom).toMatchSnapshot();

        await a11yCheck(page);
      });
    });
  });
});
