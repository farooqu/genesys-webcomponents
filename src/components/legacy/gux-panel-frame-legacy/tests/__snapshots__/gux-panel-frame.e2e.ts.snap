// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-panel-frame #render should render component as expected (1) 1`] = `"<gux-panel-frame-legacy class=\\"hydrated\\"><div slot=\\"header\\"><h1>Header</h1></div><div slot=\\"body\\"><p>Boby</p></div><div slot=\\"footer\\"><footer>footer</footer></div></gux-panel-frame-legacy>"`;

exports[`gux-panel-frame #render should render component as expected (1) 2`] = `
<section class="gux-panel-container">
  <header class="gux-panel-header">
    <slot name="header"></slot>
  </header>
  <div class="gux-panel-body">
    <slot name="body"></slot>
  </div>
  <footer class="gux-panel-footer">
    <slot name="footer"></slot>
  </footer>
</section>
`;

exports[`gux-panel-frame #render should render component as expected (2) 1`] = `"<gux-panel-frame-legacy class=\\"hydrated\\"><div slot=\\"body\\"><p>Boby</p></div><div slot=\\"footer\\"><footer>footer</footer></div></gux-panel-frame-legacy>"`;

exports[`gux-panel-frame #render should render component as expected (2) 2`] = `
<section class="gux-panel-container">
  <div class="gux-panel-body">
    <slot name="body"></slot>
  </div>
  <footer class="gux-panel-footer">
    <slot name="footer"></slot>
  </footer>
</section>
`;

exports[`gux-panel-frame #render should render component as expected (3) 1`] = `"<gux-panel-frame-legacy class=\\"hydrated\\"><div slot=\\"header\\"><h1>Header</h1></div><div slot=\\"body\\"><p>Boby</p></div></gux-panel-frame-legacy>"`;

exports[`gux-panel-frame #render should render component as expected (3) 2`] = `
<section class="gux-panel-container">
  <header class="gux-panel-header">
    <slot name="header"></slot>
  </header>
  <div class="gux-panel-body">
    <slot name="body"></slot>
  </div>
</section>
`;

exports[`gux-panel-frame #render should render component as expected (4) 1`] = `"<gux-panel-frame-legacy class=\\"hydrated\\"><div slot=\\"header\\"><h1>Header</h1></div><div slot=\\"footer\\"><footer>footer</footer></div></gux-panel-frame-legacy>"`;

exports[`gux-panel-frame #render should render component as expected (4) 2`] = `
<section class="gux-panel-container">
  <header class="gux-panel-header">
    <slot name="header"></slot>
  </header>
  <footer class="gux-panel-footer">
    <slot name="footer"></slot>
  </footer>
</section>
`;

exports[`gux-panel-frame #render should render component as expected (5) 1`] = `"<gux-panel-frame-legacy class=\\"hydrated\\"><div slot=\\"header\\"><h1>Header</h1></div></gux-panel-frame-legacy>"`;

exports[`gux-panel-frame #render should render component as expected (5) 2`] = `
<section class="gux-panel-container">
  <header class="gux-panel-header">
    <slot name="header"></slot>
  </header>
</section>
`;

exports[`gux-panel-frame #render should render component as expected (6) 1`] = `"<gux-panel-frame-legacy class=\\"hydrated\\"><div slot=\\"header\\"><h1>Header</h1></div><div slot=\\"body\\"><p>Boby</p></div><div slot=\\"invalid\\" id=\\"invalid\\"><span>Invalid</span></div><div slot=\\"footer\\"><footer>footer</footer></div></gux-panel-frame-legacy>"`;

exports[`gux-panel-frame #render should render component as expected (6) 2`] = `
<section class="gux-panel-container">
  <header class="gux-panel-header">
    <slot name="header"></slot>
  </header>
  <div class="gux-panel-body">
    <slot name="body"></slot>
  </div>
  <footer class="gux-panel-footer">
    <slot name="footer"></slot>
  </footer>
</section>
`;
